dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java

```diff
@@ -891,19 +904,36 @@ private void buildFindContainersQuery(final ContainerAPI.SearchParams searchPara
 		}
 
 		if(UtilMethods.isSet(searchParams.siteId())) {
-			query.append(" and identifier.host_inode = '");
-			query.append(searchParams.siteId()).append('\'');
+			// Use parameterized query to prevent SQL injection
+			query.append(" and identifier.host_inode = ?");
+			// Validate that siteId is a valid UUID
+			final String siteId = searchParams.siteId();
+			if (!UUIDUtil.isUUID(siteId)) {
+				throw new DotSecurityException("Invalid site ID format: " + siteId);

```

Comment:
Not sure this is always true.

Reply:
FIXED: Now uses isValidIdentifier(siteId) which specifically handles SYSTEM_HOST